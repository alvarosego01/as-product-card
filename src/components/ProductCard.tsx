

import { useProducts } from "../hooks/useProducts";
import React, { createContext } from "react";
import { InitialValues_I, ProductCardHandlers_I, ProductContextProps_I, Product_I, onChangeArgs_I } from "../interfaces/interfaces";

import styles from "../styles/styles.module.css";


export const ProductContext = createContext({

} as ProductContextProps_I)
const { Provider } = ProductContext;


export interface Props_I {
    product: Product_I;
    // children?: ReactElement | ReactElement[];
    children: (args: ProductCardHandlers_I) => JSX.Element;
    className?: string;
    style?: React.CSSProperties;
    onChange?: (args: onChangeArgs_I) => void;
    value?: number;
    initialValues?: InitialValues_I
}


export const ProductCard = ({ children, product, className, style, onChange, value, initialValues }: Props_I) => {

    const {
        counter,
        increaseBy,
        maxCount,
        isMaxCountReached,
        reset
    } = useProducts({
        onChange,
        product,
        value,
        initialValues
    });

    return (
        // Se provee la informaciÂ´pn que necesitan los hijos
        <Provider value={{
            counter,
            increaseBy,
            product,
            maxCount,
        }}>
            <div style={style} className={`${styles.productCard} ${className}`}>
                {
                    children({
                        count: counter,
                        isMaxCountReached,
                        maxCount,
                        product,
                        increaseBy,
                        reset
                    })
                }
            </div>
        </Provider>
    )

}


