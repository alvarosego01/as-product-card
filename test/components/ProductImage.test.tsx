import ProductCard, { ProductImage } from "../../src/components";
import renderer from 'react-test-renderer';
import React from 'react';
import { product2 } from "../data/products";


describe('Pruebas en ProductImage', () => {

    test('Debe mostrar el componente con imagen', () => {

        const wrapper = renderer.create(
            <ProductImage img="https://picsum.photos/100"  />
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

        expect(wrapper.root.findByType('img').props.src).toBe('https://picsum.photos/100');

    })

    test('Debe mostrar el componente con imagen del producto', () => {

        const wrapper = renderer.create(
           <ProductCard product={product2} >
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    })



})
