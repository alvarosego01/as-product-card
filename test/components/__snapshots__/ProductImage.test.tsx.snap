// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage Debe mostrar el componente con imagen 1`] = `
<img
  alt="cofee mug"
  className="productImg undefined"
  src="https://picsum.photos/100"
/>
`;

exports[`Pruebas en ProductImage Debe mostrar el componente con imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="cofee mug"
    className="productImg undefined"
    src="public/coffee-mug.png"
  />
</div>
`;
